### ISSUE TITLE
Error: fork/exec /var/task/bootstrap: no such file or directory","errorType":"Unknown"} - CustomRunTime (NET 5) 

### ISSUE THREAD

**Issue by tyler-shuhnicki** [aws/aws-lambda-dotnet#731]
Hello,

We are building and deploy lambda apps on a windows environment. While using the visual studio tools to deploy to our lambdas, everything works as expected. 

I am hooking the deploys up to a Gitlabs pipeline, and have the build machine hosted in EC2. We are using the command dotnet lambda deploy-serverless. From the deployment perspective, it is working as expected - no errors. But, when we hit the gateway, we get a 502 Bad Gateway with the following error from the function:

Tue Sep 15 11:55:21 UTC 2020 : Lambda execution failed with status 200 due to customer function error: RequestId: 0b59a014-da05-49a0-98c3-fe2ac06de4fd Error: fork/exec /var/task/bootstrap: no such file or directory. Lambda request id: 0b59a014-da05-49a0-98c3-fe2ac06de4fd
Tue Sep 15 11:55:21 UTC 2020 : Method completed with status: 502

After inspecting the deployment package, the bootstrap file is clearly present. Using a diff tool, the zip files created from visual studio vs the linux EC2 are nearly identical. The only thing I can imagine is some kind of switch that needs to be ran due to the machine being on linux. Any ideas?

Here are the msbuild parameters and framework used in the aws-lambda-tools-default.json file.

  "msbuild-parameters": "--self-contained true /p:PublishReadyToRun=true /p:PreserveCompilationContext=true",
  "framework": "net5.0",

**Comment by tyler-shuhnicki** [comment]
Throwing in some more details as I've thought about this more -

On the build machine, it is being built using NET 5. Since NET5 is in pre-release, the SDK is recommended to be installed via snap. The command is invoked by using dotnet-sdk.dotnet, as a way to avoid conflict with the current LTS release.

The pipeline runs two commands prior to running:

export DOTNET_ROOT=/snap/dotnet-sdk/current
export PATH=/snap/dotnet-sdk/current:$PATH

The idea is that any dotnet frameworks, libraries, etc. will use the NET5 frameworks. I'm curious if this is part of the issue - missing binaries or something like that.

**Comment by tyler-shuhnicki** [comment]
Issue is solved!

For anyone who might be stumbling through random links on the internet, I solved it by changing the version of the .NET preview installed on the machine.

Apparently, our project has been built on .NET5.0 version 5.0.100-preview.4.20258.7. The machine was updated with the latest version of the preview, assuming backwards compatibility. Only after noticing that the build on our local machines used msbuild 16.7 and the build machine used 16.8 did I think to track down the older version of the preview. 

I installed .NET 5.0.100-preview.4.20258.7 using the .tar.gz file, updated the path to include this destination folder (usr/share/dotnet/), and ran a command to link usr/share/dotnet to /usr/bin/dotnet. After these steps, things worked as expected.
